### PR TITLE
Correctly import ics files

### LINKED ISSUE #52: Import of ics files into Maya Google Calendar fails
Dear all, 

I cannot import ics files (see attached example) into a Google Calendar that I had added to Maya. 

My calendar entries are properly shown.  I can also manually create new events to the Google Calendar.

Anyone else experiencing problem with import of ics?

I recall that it used to work with previous versions of Maya.  I am using the latest (daily) version of Maya. 

Thanks ahead!

Robert

Launchpad Details: [#LP1444950](https://bugs.launchpad.net/bugs/1444950) Robert Orzanna - 2015-04-16 10:57:48 +0000

### LINKED ISSUE #56: Can't import ics
Elementary OS Freya

 I can't open an "ics" archive.

###########################
$ maya-calendar
[_LOG_LEVEL_INFO 10:37:38.735042] Application.vala:155: Calendar version: 0.3.1
[_LOG_LEVEL_INFO 10:37:38.735126] Application.vala:157: Kernel version: 3.16.0-38-generic

###########################
[_LOG_LEVEL_FATAL 10:37:58.477912] CalendarModel.vala:149: No calendar client was found
[_LOG_LEVEL_FATAL 10:37:58.477979] Calendar will not function properly.
*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

*** BUG ***
In pixman_region32_init_rect: Invalid rectangle passed
Set a breakpoint on '_pixman_log_error' to debug

#######################################
[_LOG_LEVEL_FATAL 10:46:34.626184] string_to_string: assertion 'self != NULL' failed
[_LOG_LEVEL_FATAL 10:46:34.626235] Calendar will not function properly.
[_LOG_LEVEL_FATAL 10:46:34.626297] [libecal] e_cal_client_create_object: assertion 'icalcomp != NULL' failed
[_LOG_LEVEL_FATAL 10:46:34.626334] Calendar will not function properly.

#######################

Events aren't in the calendar

Launchpad Details: [#LP1458143](https://bugs.launchpad.net/bugs/1458143) nh3 - 2015-05-23 08:49:55 +0000

### PR DESCRIPTION
**Pull request by Thalexander**
This fixes the import behaviour, so that it will import all VEVENTs correctly from an ics file (I had to use the libical iterators so it ended up with a pretty ugly for loop.) Previously it was taking the whole ics file as a single component for one event, which is naturally not the correct behaviour.

Fixes #56 
Fixes #52

### CODE REVIEW

**Comment by tintou** [comment]
Something is wrong with the indentation

**Comment by Thalexander** [comment]
Do you mean the for loop? because that is spread over multiple lines because it got quite long, and it i indented to line up with the start of the bracketing rather than the rest of the body to differentiate itself a bit.

**Comment by tintou** [comment]
Yeah I commented too quickly and then saw that it was a loop, everything is fine

File: src/ImportDialog.vala [hunk 1]
```diff
@@ -74,11 +74,14 @@ public class Maya.View.ImportDialog : Gtk.Dialog {
...
         foreach (var file in files) {
-            var ecal = new E.CalComponent ();
             var ical = E.Util.parse_ics_file (file.get_path ());
             if (ical.is_valid () == 1) {
-                ecal.set_icalcomponent ((owned) ical);
-                calmodel.add_event (source, ecal);
+                for (unowned iCal.Component comp = ical.get_first_component (iCal.ComponentKind.VEVENT);
+                     comp != null;
+                     comp = ical.get_next_component (iCal.ComponentKind.VEVENT)) {
+                    var ecal = new E.CalComponent.from_string  (comp.as_ical_string ());
```
Comment (tintou) [inline src/ImportDialog.vala]: double space here `from_string  (`

**Pull request merged**